exo/main.py

```diff
@@ -208,6 +219,59 @@ def clean_path(path):
         path = path.strip('Optional("').rstrip('")')
     return os.path.expanduser(path)
 
+async def hold_outstanding(node: Node):
+  while True:
```

Comment:
Why not simplify the condition with `while node.outstanding_request:`

Reply:
I vaguely remember trying that and being confused that it didn't work, but I can double-check. Python async has some quirks I'm not completely clear on at times

Reply:
Makes sense.
Python asyncio definitely has its quirks. I found this resource helpful: https://realpython.com/async-io-python/

Reply:
Okay, pushed a fix for this